step-capstone/src/components/Pages/TripList.js

```diff
@@ -57,43 +64,85 @@ class TripList extends React.Component {
   }
 
   handleDeleteTrip(tripId) {
-    this.context.deleteTrip(this.state.reference+tripId).then(() => this.loadTrips());
+    this.context.deleteTrip(this.state.reference + tripId).then(() => this.loadTrips());
     this.loadTrips();
   }
 
+  handleEditTripClick(tripId) {
+    this.context.getTrip(this.state.reference + tripId)
+      .then((tripSetting) => {
+        tripSetting = tripSetting.data()
+        tripSetting.startDate = tripSetting.startDate.toDate()
+        tripSetting.endDate = tripSetting.endDate.toDate()
+        this.setState({
+          selectedTripSetting: tripSetting,
+          selectedTripId: tripId,
+          editingTrip: true
+        });
+      })
+  }
+
+  handleEditTripSetting(newSetting) {
+    this.context.editTripSetting(this.state.reference + this.state.selectedTripId, this.state.selectedTripSetting, newSetting)
+      .then(() => { console.log("load trips"); this.loadTrips() })
```

Comment:
Nit: Please make the log message more specific or remove.

Reply:
removed, thanks